pkg/util/metric/histogram_buckets.go

```diff
@@ -0,0 +1,178 @@
+	12247448713.915894,  // 12.247448713s
+	19341270753.704967,  // 19.341270753s
+	30543892291.876068,  // 30.543892291s
+	48235163460.447227,  // 48.23516346s
+	76173362969.685760,  // 1m16.173362969s
+	120293595166.717728, // 2m0.293595166s
+	189968625172.725128, // 3m9.968625172s
+	300000000000.000183, // 5m0s
+}
+
+// LongRunningProcessLatencyBuckets are prometheus histogram buckets suitable
+// for a histogram that records a quantity (nanosecond-denominated) for
+// long-running processes (multiple minutes).
+var LongRunningProcessLatencyBuckets = []float64{
+	// Generated via TestHistogramBuckets/LongRunningProcessLatencyBuckets.
+	500000000.000000,     // 500ms
+	942961049.923126,     // 942.961049ms
+	1778351083.344248,    // 1.778351083s
+	3353831609.364442,    // 3.353831609s
+	6325065151.263324,    // 6.325065151s
+	11928580151.734879,   // 11.928580151s
+	22496372927.944168,   // 22.496372927s
+	42426406871.192848,   // 42.426406871s
+	80012898335.451462,   // 1m20.012898335s
+	150898093243.579315,  // 2m30.898093243s
+	284582048872.726685,  // 4m44.582048872s
+	536699575188.601318,  // 8m56.699575188s
+	1012173589826.278687, // 16m52.173589826s
+	1908880541934.094238, // 31m48.880541934s
+	3599999999999.998535, // 59m59.999999999s
+}
+
+// CountBuckets are prometheus histogram buckets suitable for a histogram that
+// records a quantity that is a count (unit-less) in which most measurements are
+// in the 1 to ~1000 range during normal operation.
+var CountBuckets = []float64{
+	// Generated via TestHistogramBuckets/CountBuckets.
+	1.000000,
+	2.000000,
+	4.000000,
+	8.000000,
+	16.000000,
+	32.000000,
+	64.000000,
+	128.000000,
+	256.000000,
+	512.000000,
+	1024.000000,
+}
+
+// PercentBuckets are prometheus histogram buckets suitable for a histogram that
+// records a percent quantity [0,100]
+var PercentBuckets = []float64{
+	// Generated via TestHistogramBuckets/PercentBuckets.
+	10.000000,
+	20.000000,
+	30.000000,
+	40.000000,
+	50.000000,
+	60.000000,
+	70.000000,
+	80.000000,
+	90.000000,
+	100.000000,
+}
+
+// DataSizeBuckets are prometheus histogram buckets suitable for a histogram that
+// records a quantity that is a size (byte-denominated) in which most measurements are
+// in the kB to MB range during normal operation.
+var DataSizeBuckets = []float64{
+	// Generated via TestHistogramBuckets/DataSizeBuckets.
+	1000.000000,     // 1.0 kB
+	2000.000000,     // 2.0 kB
+	4000.000000,     // 4.0 kB
+	8000.000000,     // 8.0 kB
+	16000.000000,    // 16 kB
+	32000.000000,    // 32 kB
+	64000.000000,    // 64 kB
+	128000.000000,   // 128 kB
+	256000.000000,   // 256 kB
+	512000.000000,   // 512 kB
+	1024000.000000,  // 1.0 MB
+	2048000.000000,  // 2.0 MB
+	4096000.000000,  // 4.1 MB
+	8192000.000000,  // 8.2 MB
+	16384000.000000, // 16 MB
+}
+
+// MemoryUsageBuckets are prometheus histogram buckets suitable for a histogram that
+// records memory usage (in Bytes)
+var MemoryUsageBuckets = []float64{
```

Comment:
ditto an upper limit of 19kb seems very arbitrary and not sufficient for many future memory monitoring asks. For example, in the raft transport we're dealing with entries that are SSTs, typically of size 16mb, so there we'd want at least the 16mb range to be resolved properly.